Implemented WAR segregated server interceptors

ronsigal opened the pull request.

jamezp commented on subsystem/src/main/java/org/wildfly/extension/grpc/Priority.java:
```diff
@@ -0,0 +1,32 @@
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface Priority {
```
Any reason not to use `jakarta.annotation.Priority`?

ronsigal replied:
Actually, I was having trouble because the @Priority annotation was recognized, so I tried writing my own. Then I added jandex and the problem went away. I guess I should revert to `jakarta.annotation.Priority`.

ronsigal replied:
Fixed

ronsigal commented on subsystem/pom.xml:
```diff
@@ -171,6 +171,23 @@
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.jboss.jandex</groupId>
+                <artifactId>jandex-maven-plugin</artifactId>
```
Oh, ok.

ronsigal replied:
Done

ronsigal commented on subsystem/src/main/java/org/wildfly/extension/grpc/InterceptorQueue.java:
```diff
@@ -0,0 +1,115 @@
+
+import io.grpc.ServerInterceptor;
+
+/**
+ * Specialized {@link PriorityQueue} that sorts {@link ServerInterceptor}s based on the value of a
+ * {@link @Priority} annotation. A {@link ServerInterceptor} with no {@link @Priority} annotation has
+ * the lowest priority.
+ */
```
Ok. It's meant for internal use, but even so, ok.

ronsigal replied:
Done

ronsigal commented:
I've removed the "Work in Progress" label because I don't know of anything else that needs to be done now.

Well, do we need to do something for the Preview stability level?

ronsigal commented:
Another thing: the tests are failing with "To build this project JDK 17 (or greater) is required. Please install it." 

I'm not sure how to do that.

darranl commented:
@ronsigal I am not quite following what has happened to the subsystem schema in this PR as it looks to include a lot of unrelated changes.

Also we seem to have a new .elements file but I am not sure if this is a side effect of a client you are using to edit the schema.

ronsigal commented:
> @ronsigal I am not quite following what has happened to the subsystem schema in this PR as it looks to include a lot of unrelated changes.

@darranl At some point in the past, I added all of those parameters. Now I've added them to the schema.

@darranl 
> 
> Also we seem to have a new .elements file but I am not sure if this is a side effect of a client you are using to edit the schema.

Ah, I was concerned that having so many attributes just didn't look nice in the xml file, so I started looking at turning them into elements. Then someone argued that it doesn't really matter what the xml file looks like, so I dropped it. Looks like I accidentally committed the file, though. I'll get rid of it.

hpehl reviewed (commented):
I can't really evaluate this PR. I don't know enough about subsystem programming for that.

darranl commented:
@hpehl / @jamezp I would suggest we don't go too deep on the review of this one if generally Ok

We just discussed using 0.1.4 as the next release and we are going to iterate for a bit before we jump to 0.2.0 so not a problem / blocker if we merge this and continue to evolve.
